### PR TITLE
Enabled monitoring 

### PR DESCRIPTION
**Pull request by maltesander**
- Added should_be_scraped annotation
- Added finalizer
- Added container port
- Adapted to product_config_path from operator-rs

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)

### CODE REVIEW

File: operator/src/lib.rs [hunk 1]
```diff
@@ -293,11 +297,17 @@ impl OpaState {
...
+
+        let mut annotations = BTreeMap::new();
+        // only add metrics container port and annotation if available
+        if let Some(metrics_port) = port {
+            annotations.insert(SHOULD_BE_SCRAPED.to_string(), "true".to_string());
+            container_builder.add_container_port(
+                ContainerPortBuilder::new(metrics_port.parse()?)
+                    .name("metrics")
+                    .build(),
+            );
```
Comment (razvan) [inline operator/src/lib.rs:310]: What about the "client" container port ?

**Review by razvan — COMMENTED** [review]
See one comment.

File: operator/src/lib.rs (hunk 1 above)
Comment (maltesander) [reply operator/src/lib.rs:310]: They are identical.

**Review by razvan — APPROVED** [review]
ok

**Pull request merged**